### ISSUE TITLE
Left hand - fingers abduction does not move

### ISSUE THREAD

**Issue by plinioMoreno** [robotology/icub-tech-support#1474]
### Device name 🤖

iCubLisboa01

### Request/Failure description

The left hand fingers abduction joint (`l_hand_finger`) does not move after starting the robot. If we try to move the motor on `yarpmotorgui` the motor does not move, stays at the same location. 

### Detailed context

The following video shows the `yarpmotorgui` screen:

https://user-images.githubusercontent.com/3798890/205710153-a75cd754-1b7c-4ef4-83ba-43e6668dfd70.mp4

The log from `yarprobotinterface` as follows:

[log_icub-head_yarprobotinterface_3000.txt](https://github.com/robotology/icub-tech-support/files/10156741/log_icub-head_yarprobotinterface_3000.txt)

Please ignore the errors from the right arm, which are part of other issue.

### How does it affect you?

We cannot run grasping experiments with the hands.

**Comment by plinioMoreno** [comment]
Hi @AntonioConsilvio,

We removed the cover of the left hand to see if there was something wrong with this joint. After placing the hand cover back, now the joint (`l_hand_finger`) goes to fault during startup. The log from `yarprobotinterface` as follows:

 
[log_icub-head_yarprobotinterface_3279.txt](https://github.com/robotology/icub-tech-support/files/10177699/log_icub-head_yarprobotinterface_3279.txt)

Let us know how to fix this problem.

cc @Ryo-Naohara

**Comment by AntonioConsilvio** [comment]
Hi @plinioMoreno, ok this shouldn't be an electrical problem. 
So, try to remove the back of the hand and with a slotted screwdriver try to manually move the motor (green, in the photo below).
![Adduzione1](https://user-images.githubusercontent.com/114915464/207098645-415fd252-4a01-4cf1-82f5-e74be9b7de80.jpg)

If, when you move the motor with the screwdriver, fingers don't move it means there is a problem with the steel cable (whose condition you can check by following the blue arrow in the picture above) or with the motor.

cc @Fabrizio69

**Comment by plinioMoreno** [comment]
Hi @AntonioConsilvio, this is the video that shows that the fingers move correctly.


https://user-images.githubusercontent.com/3798890/208464702-5b1db110-2e32-483d-bb54-f81eb6d49d77.mp4


It seems that the cable is fine. However, we still have the behavior described [here](https://github.com/robotology/icub-tech-support/issues/1474#issue-1477133521)

**Comment by AntonioConsilvio** [comment]
Hi @plinioMoreno, I am not sure about the cause of the problem, but you can try checking the connection of the Hall sensor (located above the little finger, blue), whose cable (orange) go to the EB24 board (green).




![Image](https://user-images.githubusercontent.com/114915464/210211029-3b11b3cc-7223-4d74-92c2-9062ea628e64.jpg)


(The photo above is only for visual inspection, do not touch these cables if they appear as in the photo above)

![Image](https://user-images.githubusercontent.com/114915464/210211214-2ffe3d70-62a9-4297-9424-4596f239265a.jpg)


To find the cable, you need to check the colored cubes on the cable and find the cable called 2M3. The EB24 board is the first of the three board present in the left arm.



![Image](https://user-images.githubusercontent.com/114915464/210211465-de395941-091a-44e2-9a4a-0530f5ceb918.jpg)


Write to us if you have any questions. 

cc @Fabrizio69

**Comment by AntonioConsilvio** [comment]
Hi @plinioMoreno, I propose that you schedule a videocall on Monday, February 23 at 10:00 a.m. in your time zone.

Are you available on this date?

cc @Fabrizio69

**Comment by plinioMoreno** [comment]
Hi @AntonioConsilvio, that sounds good. I confirm that @Ryo-Naohara is available to talk with you on the coming monday. Do we need to setup something on our icub laptop?

**Comment by AntonioConsilvio** [comment]
> Do we need to setup something on our icub laptop?
@plinioMoreno This should not be necessary, just have the laptop ready to start the robot. I suggest keeping the back of the robot's hand and the left upper-arm cover open.

**Comment by AntonioConsilvio** [comment]
Hi @plinioMoreno, after the video call we suspect that the motor needs to be replaced. 
So, analyzing the situation, and the other numerous problems found on the left hand, we wanted to suggest that you send us the robot arm so that we can fix all the problems present on the hand. 

If you agree, I will show you how to separate the arm from the robot.

cc @Fabrizio69

**Comment by plinioMoreno** [comment]
Hi @AntonioConsilvio,

Do we have to remove the full arm? Or just the forearm? I am asking this because we also found some missing tactile information on the right hand.

**Comment by AntonioConsilvio** [comment]
@plinioMoreno I'm sorry, I didn't quite understand. You should send us almost the whole arm. You should remove those 4 screws (red) and remove some connectors from the electronics boards. I can tell you which connectors if you want to send us the left arm.



![Image](https://user-images.githubusercontent.com/114915464/215049220-c10d9ac4-60fa-4582-b793-cb235e1bc399.png)


However, I did not understand why you are talking about the right hand, do you have problems with the right hand as well? If you have problems with it, what problems do you have?

Finally, I did not understand whether you agree to send us the left arm or not. 

Basically, I would need more precise information. Thank you.

cc @Fabrizio69

**Comment by plinioMoreno** [comment]
Hi @AntonioConsilvio,

Sorry for the multiple information. Regarding the right hand, yes, we are having problems with the taxels of the thumb, where the values do not change in the skinGui. Ryo will open an issue to see if we solve that problem with your help.

Regarding the left hand/arm, we agree with sending you the arm during next week. We will remove the arm and send it to you.

**Comment by AntonioConsilvio** [comment]
Hi @plinioMoreno, before activating shipping, perhaps it may be helpful to know the right arm problem, to see if it may be smart to send us both front arms, or if the right arm problem can be solved with remote support.

cc @Fabrizio69

**Comment by plinioMoreno** [comment]
Hi @AntonioConsilvio,

The right arm problem is described in [this issue](https://github.com/robotology/icub-tech-support/issues/1493#issue-1562550875).

**Comment by plinioMoreno** [comment]
Hi @AntonioConsilvio, after the final diagnostic of the right hand [here](https://github.com/robotology/icub-tech-support/issues/1493#issue-1562550875), we are going to send just the left arm. @Ryo-Naohara unmounted the arm and we are ready to send it to you. Do you want to send the address and sender info please?

**Comment by andreaderito** [comment]
Hi @plinioMoreno the address where to send the parcel is:
IIT - Istituto Italiano di Tecnologia
Via Morego 30
I-16163 Genoa, ITALY
Att.n: Fabrizio Larosa (iCub Tech, CIRS)
Phone: +39 010 28 98 223

**IMPORTANT: please safely pack the goods using the maximum care to absorb shocks**. If possible, indicate "**fragile**" on the parcel. As soon as you have a **tracking number**, pls let us know it so we may follow the shipping. **If possible, ensure the goods with the shipping company** - indicative (reasonable) values you may put 30keur for 1 lower-arm (forearm, wrist and hand altogether).
If you need further info, pls let me know.

cc @Fabrizio69 @AntonioConsilvio @maggia80

**Comment by andreaderito** [comment]
Ok, thank you. Just in case you have a tracking number, pls let me have it and if it is DHL or TNT or other - if you don't have it handy, no problem of course - thx and best.

**Comment by plinioMoreno** [comment]
I do not have tracking number info, but this is the picture:
![imagem](https://user-images.githubusercontent.com/3798890/223501429-a6603b36-9c55-45eb-81ef-e83ce590d577.png)

**Comment by andreaderito** [comment]
Hi Plinio, we received it - in the next day (or days, as soon as my colleagues are available), we'll open it and start working on it. Thanks!
cc @Fabrizio69 @maggia80 @AntonioConsilvio

**Comment by plinioMoreno** [comment]
Hi @andreaderito,

Good that arrived safely, and thanks for letting us know!

**Comment by AntonioConsilvio** [comment]
Hi @plinioMoreno, we opened the package and we noticed that the directions given in this [comment](https://github.com/robotology/icub-tech-support/issues/1474#issuecomment-1406230197) were not followed. 

It means that we haven't the electronics boards and that the pronosupination mechanism has been split in half.

It create two problems:
1. After the interventions, we cannot fully test the whole assembly, but only partially because we haven't the electronics boards.
2.  The way the pronosupination mechanism was disassembled was not according to the instructions provided, so the harmonic drive was damaged, and it needs to be replaced.

We show an example of a our ForeArm:
![ForeArm](https://user-images.githubusercontent.com/114915464/225293794-dde255f9-6d60-482e-ac55-02149117cc17.png)

Your ForeArm shows like this:
![ForeArm Lisbona](https://user-images.githubusercontent.com/114915464/225293963-f40af494-7f54-44c8-9b78-8a93180d7084.png)

So We need this mechanical part (RED in the photo below) that you should have. Because we have to change the harmonic drive and riassemble the Pronosupination mechanism.

![ForeArm2](https://user-images.githubusercontent.com/114915464/225295528-81f4d1ba-63b1-4d26-a237-f62f33d0a6a1.jpg)


Since you will send this mechanical part, you should also send the electrical boards, the following the instructions in this [comment](https://github.com/robotology/icub-tech-support/issues/1474#issuecomment-1406230197).


 If the instructions are not clear or if you have any question, please write us.

cc @sgiraz @Fabrizio69 @maggia80

**Comment by plinioMoreno** [comment]
Hi @AntonioConsilvio,

First, we are sorry for the misunderstanding. By mistake @Ryo-Naohara thought that the screws to be removed are the ones marked with black circles:
![missing_parts](https://user-images.githubusercontent.com/3798890/225320473-bef621a8-d339-4706-abd5-b5e3fb73fd1c.jpg)

We want to double check with you that the missing parts are the ones in the picture above (control boards EB24,EB25 and EB26). Then, we will let you know when the package is sent.

Also we want to understand why the harmonic drive got damaged, since all the parts seemed to be okay when removed.

**Comment by AntonioConsilvio** [comment]
@plinioMoreno, Yes, the electronics boards are the ones in your photo.

Regarding the harmonic transmission, the Pronosupination mechanism has a specific sequence of assembly. So, consequently, it has a specific sequence of disassembly and in that sequence that mechanical part is the last one that has to be removed.


So, probably the harmonic transmission is probably damaged (we hope not) and to verify that we need that part, because even if it is not visibly damaged, we need to verify it to be sure.

cc @Fabrizio69 @sgiraz

**Comment by Ryo-Naohara** [comment]
@AntonioConsilvio, is this the steel plate?
![IMG_20230315_155543](https://user-images.githubusercontent.com/119933065/225384496-48a9a7f2-994d-438d-a5d5-18a9da4849d9.jpg)

**Comment by AntonioConsilvio** [comment]
@Ryo-Naohara, Yes it is the steel plate

**Comment by plinioMoreno** [comment]
Hi @andreaderito and @AntonioConsilvio,

The missing parts of the left arm, control boards and steel plate were shipped yesterday through DHL. @andreaderito should have received an email, and the small box should arrive today to your facilities.

**Comment by andreaderito** [comment]
Hi @plinioMoreno thanks, we'll inform you when the parcel arrives (no email up to now, but  absolutely no problem, we know the parcel is about to arrive so we just wait for it - thx).
cc @AntonioConsilvio

**Comment by GiacomoVassallo13** [comment]
Good morning. the abduction joint motor was replaced and the tendon rewired. Once reconnected it will be electrically tested. 

![IMG_20230323_140424](https://user-images.githubusercontent.com/70649516/227215422-58e14d87-74ce-4f85-bd2b-979be0b425af.jpg)

cc: @sgiraz @Fabrizio69 @AntonioConsilvio

**Comment by AntonioConsilvio** [comment]
Hi @andreaderito @plinioMoreno, the parcel has arrived and, at first glance, everything looks fine.

cc @sgiraz @Fabrizio69

**Comment by sgiraz** [comment]
Hi @plinioMoreno,

## ⚙️ Maintenance 
We finalized all the activities on the left hand of `iCubLisbona01`. Check out the video below to see it in action:

| Startup  | Adduction |
| ------------- | ------------- |
| <video src="https://user-images.githubusercontent.com/114915464/235850884-13545cf3-4ba0-4a7c-9148-2034d5b60f23.mp4">  | <video src="https://user-images.githubusercontent.com/114915464/235850890-968eb69e-4196-48e4-8f56-0e0d6caeae3b.mp4"> |


| Skin test |
| ------------- |
| <video src="https://user-images.githubusercontent.com/114915464/235710142-c35694b0-82f8-44ab-885b-b174f2b8878a.mp4">  |

--- 

## 📦 Shipment 
As soon as it is shipped, you will receive the following package:
 
| Left hand | Fingertips + Spare wire | Full package |
| ------------- | ------------- | ------------- | 
| ![image](https://github.com/robotology/icub-tech-support/assets/18039045/e4163084-156b-4549-9fcf-63133a503117) | ![image](https://github.com/robotology/icub-tech-support/assets/18039045/d61df6c2-62ca-42c9-8638-60b615af0841) | ![image](https://github.com/robotology/icub-tech-support/assets/18039045/4d1408a5-7390-4c39-9c5c-6423a2b90cad) |

> **Note**
> notes here

**Comment by sgiraz** [comment]
I leave this issue in `review`.

@plinioMoreno let us know when you received the package.

**Comment by andreaderito** [comment]
Hi @plinioMoreno I've acted so as to ask for the Documento di Trasporto (DDT, valid only in Italy) and also prepared the Packing List with the value of the goods (this file is already finalised, so pls find it here since now in case you wish starting preparing the insurance coverage of the goods). **The goods will be ready for pick up since next Tuesday, May 16th, 2023 at 08.30 from the address of IIT Central Warehouse i.e. IIT - Via Morego 30, 16163 Genoa, Italy**. Pls let us know when the goods arrive at your premises so we relax a bit - best, Andrea

[PACKING LIST (DETAILED_LIST_OF_GOODS)_v1_DEF.pdf](https://github.com/robotology/icub-tech-support/files/11472482/PACKING.LIST.DETAILED_LIST_OF_GOODS._v1_DEF.pdf)

cc @sgiraz @AntonioConsilvio @Fabrizio69 @maggia80

**Comment by plinioMoreno** [comment]
@sgiraz ,thank you for the pictures and the videos. From them I see that everything looks good, thanks!

@andreaderito, if we want to do the insurance, how shall we proceed?

**Comment by andreaderito** [comment]
Hi @plinioMoreno  normally you may select this option when you book the express courier (DHL or similar, pls check with your admin if they know how to do it - otherwise pls let me know, best

**Comment by andreaderito** [comment]
Hi @Plinio, a small but important errata corrige: you may book the pick up of the goods starting from Thursday 18th morning (not from today, sorry). The pick up address is IIT Central Warehouse, Via Morego 30, 16163 Genoa, Italy (no change in respect to what indicated before). Best.
cc @Miche19

**Comment by AntonioConsilvio** [comment]
Hi @plinioMoreno, remember to update the robots-configuration files, because I update the calibration of some joints after the maintenance service.

- https://github.com/robotology/robots-configuration/pull/501

cc @sgiraz

**Comment by plinioMoreno** [comment]
We put the left arm back and everything is working as expected. Thanks @AntonioConsilvio for the reminder on the configuration files. From our side, this issue can be closed.

Also thanks to all the people involved in the repair.
